### PR TITLE
When partitioning fails, show the actual problem

### PR DESCRIPTION
**Pull request by jimmykarily**

### CODE REVIEW

File: pkg/partitioner/disk.go [hunk 1]
```diff
@@ -17,13 +17,14 @@ limitations under the License.
...
 
 import (
-	"errors"
 	"fmt"
 	"os"
 	"regexp"
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
```
Comment (mudler) [inline pkg/partitioner/disk.go:26]: this has been archived

File: pkg/partitioner/disk.go [hunk 2]
```diff
@@ -119,7 +120,7 @@ func (dev *Disk) Reload() error {
 	pc := NewPartedCall(dev.String(), dev.runner)
 	prnt, err := pc.Print()
 	if err != nil {
-		return err
+		return errors.Wrap(err, prnt)
```
Comment (mudler) [inline pkg/partitioner/disk.go:123]:
```suggestion
		return fmt.Errorf("error: %w. output: %s",err, prnt)
```

File: pkg/partitioner/disk.go [hunk 3]
```diff
@@ -131,14 +132,14 @@ func (dev *Disk) Reload() error {
 	if unallocatedRegexp.Match([]byte(prnt)) {
 		// Parted has not a proper way to doing it in non interactive mode,
 		// because of that we use sgdisk for that...
-		_, err = dev.runner.Run("sgdisk", "-e", dev.device)
+		out, err := dev.runner.Run("sgdisk", "-e", dev.device)
 		if err != nil {
-			return err
+			return errors.Wrap(err, string(out))
```
Comment (mudler) [inline pkg/partitioner/disk.go:137]:
```suggestion
			return  fmt.Errorf("error: %w. output: %s",err, string(out))
```

File: pkg/partitioner/disk.go [hunk 4]
```diff
@@ -131,14 +132,14 @@ func (dev *Disk) Reload() error {
...
 		}
 		// Reload disk data with fixed headers
 		prnt, err = pc.Print()
 		if err != nil {
-			return err
+			return errors.Wrap(err, prnt)
```
Comment (mudler) [inline pkg/partitioner/disk.go:142]:
```suggestion
			return  fmt.Errorf("error: %w. output: %s",err, prnt)
```

File: pkg/partitioner/disk.go (hunk 1 above)
Comment (jimmykarily) [reply pkg/partitioner/disk.go:26]: oh! I didn't know that. Will use `fmt.Errorf` or something. Preparing a fix

File: go.sum [hunk 5]
```diff
@@ -560,7 +560,6 @@ github.com/docker/go-metrics v0.0.0-20180209012529-399ea8c73916/go.mod h1:/u0gXw
 github.com/docker/go-metrics v0.0.1 h1:AgB/0SvBxihN0X8OR4SjsblXkbMvalQ8cjmtKQ2rQV8=
```
Comment (jimmykarily) [inline go.sum:560]: I don't know why `go mod tidy` made these changes :man_shrugging:

**Review by mudler — APPROVED** [review]

**Pull request merged**